device/globals/bg_music.gd

```diff
@@ -7,8 +7,8 @@ func game_cleared():
 	if global_id != "":
 		vm.register_object(global_id, self)
 
-func set_state(p_state, p_force = false):
-
+func set_state(p_state, p_volume = 1, p_force = false):
```

Comment:
As you found out and we discussed on IRC, this second parameter to `set_state` is problematic, because the state is not saved. I think the idiomatic, and most pragmatic, solution might be to use `set_global` to set volume and add a listener for the `global_changed` event, which you'll find examples of in eg. `item.gd` and `inventory.gd`.